internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,357 @@
+package datasource
+
+import (
+	"errors"
+	"github.com/DATA-DOG/go-sqlmock"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/vatsal278/TransactionManagementService/internal/config"
+	"github.com/vatsal278/TransactionManagementService/internal/model"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestSqlDs_HealthCheck(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping testing due to unavailability of testing environment")
+	}
+	//include a failure case
+	dbcfg := config.DbCfg{
+		Port:      "9085",
+		Host:      "localhost",
+		Driver:    "mysql",
+		User:      "root",
+		Pass:      "pass",
+		DbName:    "useracc",
+		TableName: "newTemp",
+	}
+	dataBase := config.Connect(dbcfg, dbcfg.TableName)
+	svcConfig := config.SvcConfig{
+		DbSvc: config.DbSvc{Db: dataBase},
+	}
+	dB := NewSql(config.DbSvc(svcConfig.DbSvc), "newTemp")
+
+	tests := []struct {
+		name        string
+		setupFunc   func()
+		cleanupFunc func()
+		filter      map[string]interface{}
+		validator   func(bool)
+		dbInterface DataSourceI
+	}{
+		{
+			name: "SUCCESS::Health check",
+			validator: func(res bool) {
+				if res != true {
+					t.Errorf("Want: %v, Got: %v", true, res)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			res := dB.HealthCheck()
+
+			if tt.validator != nil {
+				tt.validator(res)
+			}
+		})
+	}
+}
+func TestGet(t *testing.T) {
```

Comment:
Test case names does not explain what is being tested. Update.

Reply:
done